File: BE/order-service/src/main/java/com/vietnl/orderservice/domain/models/entities/Order.java
package com.vietnl.orderservice.domain.models.entities;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UuidGenerator;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "orders")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Order {

    @Id
    @GeneratedValue
    @UuidGenerator
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;

    @Column(name = "table_id")
    private String tableId;

    @Column(name = "table_number")
    private String tableNumber;

    // Nhân viên tạo đơn
    @Column(name = "created_by")
    private String createdBy;

    // Trạng thái: PENDING, CONFIRMED, COMPLETED, CANCELLED
    @Column(name = "status", nullable = false)
    private String status = "PENDING";

    @Column(name = "note", length = 500)
    private String note;

    // Tổng tiền (tính lại khi thêm/xóa món)
    @Column(name = "total_amount", precision = 18, scale = 2)
    private BigDecimal totalAmount = BigDecimal.ZERO;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @OneToMany(mappedBy = "order", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    private List<OrderItem> items = new ArrayList<>();

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public void recalculateTotal() {
        this.totalAmount = items.stream()
                .map(item -> item.getUnitPrice().multiply(BigDecimal.valueOf(item.getQuantity())))
                .reduce(BigDecimal.ZERO, BigDecimal::add);
    }
}

File: FE/mobile/app/src/main/java/com/food/order/data/model/Order.kt
package com.food.order.data.model

import com.google.gson.annotations.SerializedName
import java.io.Serializable

data class Order(
    @SerializedName("id") val id: String,
    @SerializedName("tableId") val tableId: String? = null,
    @SerializedName("server") val server: String? = null,
    @SerializedName("createdAt") val createdAt: String? = null,
    @SerializedName("createdBy") val createdBy: String? = null,
    @SerializedName("status") val status: String? = null,
    @SerializedName("items") val items: List<OrderItem> = emptyList(),
    @SerializedName("totalAmount") val totalAmount: Double? = null,
    @SerializedName("employeeName") val employeeName: String? = null,
    @SerializedName("tableNumber") val tableName: String? = null
) : Serializable
